graphql_compiler/query_formatting/cypher_formatting.py

```diff
@@ -0,0 +1,134 @@
+        raise GraphQLInvalidArgumentError(u'Attempting to represent a non-list as a list: '
+                                          u'{}'.format(argument_value))
+
+    components = (
+        _safe_cypher_argument(stripped_type, x)
+        for x in argument_value
+    )
+    return u'[' + u','.join(components) + u']'
+
+
+def _safe_cypher_argument(expected_type, argument_value):
+    """Return a Cypher string representing the given argument value."""
+    if GraphQLString.is_same_type(expected_type):
+        return _safe_cypher_string(argument_value)
+    elif GraphQLID.is_same_type(expected_type):
+        # IDs can be strings or numbers, but the GraphQL library coerces them to strings.
+        # We will follow suit and treat them as strings.
+        if not isinstance(argument_value, six.string_types):
+            if isinstance(argument_value, bytes):  # should only happen in py3
+                argument_value = argument_value.decode('utf-8')
+            else:
+                argument_value = six.text_type(argument_value)
+        return _safe_cypher_string(argument_value)
+    elif GraphQLFloat.is_same_type(expected_type):
+        return represent_float_as_str(argument_value)
+    elif GraphQLInt.is_same_type(expected_type):
+        # Special case: in Python, isinstance(True, int) returns True.
+        # Safeguard against this with an explicit check against bool type.
+        if isinstance(argument_value, bool):
+            raise GraphQLInvalidArgumentError(u'Attempting to represent a non-int as an int: '
+                                              u'{}'.format(argument_value))
+        return type_check_and_str(int, argument_value)
+    elif GraphQLBoolean.is_same_type(expected_type):
+        return type_check_and_str(bool, argument_value)
+    elif GraphQLDecimal.is_same_type(expected_type):
+        return _safe_cypher_decimal(argument_value)
+    elif GraphQLDate.is_same_type(expected_type):
+        return _safe_cypher_date_and_datetime(expected_type, (datetime.date,), argument_value)
+    elif GraphQLDateTime.is_same_type(expected_type):
+        return _safe_cypher_date_and_datetime(expected_type,
+                                              (datetime.datetime, arrow.Arrow), argument_value)
+    elif isinstance(expected_type, GraphQLList):
+        return _safe_cypher_list(expected_type.of_type, argument_value)
+    else:
+        raise AssertionError(u'Could not safely represent the requested GraphQL type: '
+                             u'{} {}'.format(expected_type, argument_value))
+
+
+######
+# Public API
+######
+
+
+def insert_arguments_into_cypher_query(compilation_result, arguments):
+    """Insert the arguments into the compiled Cypher query to form a complete query.
+
+    Args:
+        compilation_result: CompilationResult, compilation result from the GraphQL compiler.
+        arguments: Dict[str, Any], parameter name -> value, for every parameter the query expects.
```

Comment:
I've usually seen ```dict, str -> any, mapping x to y``` in docstrings. Is there a reason why this is different?

Reply:
There doesn't seem to be any one format that's used everywhere-- for uniformity I'll change this so it matches the description in gremlin_ and match_formatting.py

Reply:
and add types to the comment as well.